### ISSUE TITLE
Pyright incorrectly reports "cannot assign to attribute" in large files

### ISSUE THREAD

**Issue by prokie** [microsoft/pyright#9875]
**Describe the bug**
When analyzing a large auto-generated Python file with many classes containing property getters and setters, Pyright incorrectly reports an assignment error for property attributes.

**Code or Screenshots**
To generate the classes you can use this code.

```python
NUM_CLASSES = 2048 # Adjust this number as needed
num_attributes = 3

def generate_class(index: int) -> str:
    attributes = '\n'.join([f'        self._attr{i} = 0' for i in range(num_attributes)])
    properties = '\n'.join([
        f'''    @property
    def attr{i}(self) -> int:
        """Getter for attr{i}"""
        return self._attr{i}

    @attr{i}.setter
    def attr{i}(self, val: int) -> None:
        """Setter for attr{i}"""
        self._attr{i} = val
''' for i in range(num_attributes)])
    
    return f'''
class Class{index}:
    """Auto-generated class {index} with {num_attributes} attributes"""
    
    def __init__(self):
{attributes}

{properties}
'''

def main():
    with open("generated_classes.py", "w") as f:
        f.write("""# Auto-generated file to stress test Pyright\n\n""")
        for i in range(NUM_CLASSES):
            f.write(generate_class(i))

if __name__ == "__main__":
    main()
```

Then create another file that uses any of the generated classes.

```python
from generated_classes import Class0

class0 = Class0()
class0.attr0 = 3  # Causes Pyright error if NUM_CLASSES > 2047
```

If you use 2048 or more for `NUM_CLASSES` you will see this error

```
Cannot assign to attribute "attr0" for class "Class0"
  Type "Literal[3]" is not assignable to type "(val: int) -> None"
```

If you use 2047 or less pyright will return no warnings.

**VS Code extension or command-line**
Same error in both

**Comment by erictraut** [comment]
This is by design. If you open the "generated_classes.py" file, you'll notice that pyright emits an error at the top that indicates that the "Code is too complex to analyze". I'll note that this code file is nearly 80000 lines in length!

There will always be some limit beyond which static analysis isn't feasible. For hand-written code, it's difficult to reach this limit. but with generated code, it's possible to do so.

Keep in mind that static type checkers need to analyze all possible paths through a code flow graph. If that code flow graph is too complex, analyzing the code will take minutes, hours or days to complete. Pyright computes an internal "complexity score" for each execution context (i.e. each function and module body) to determine whether it's feasible to perform analysis. If it's greater than a certain complexity score, it doesn't attempt to analyze the code in that context.

If you want your code to work with pyright, you'll need to break it up into submodules. Since it's generated programmatically, this should be relatively easy to do.
